core/transformer/src/IModelExporter.ts

```diff
@@ -114,11 +123,13 @@ export abstract class IModelExportHandler {
    */
   public shouldExportSchema(_schemaKey: SchemaKey): boolean { return true; }
 
-  /** Called when an schema should be exported.
+  /** Called when a schema should be exported.
    * @param schema The schema to export
    * @note This should be overridden to actually do the export.
+   * @note return an [[ExportSchemaResult]] with a `schemaPath` property to notify overrides that call `super`
+   *       where a schema was written for import.
    */
-  public async onExportSchema(_schema: Schema): Promise<void> { }
+  public async onExportSchema(_schema: Schema): Promise<void | ExportSchemaResult> { }
```

Comment:
Motivation for using `void | ExportSchemaResult` instead of more idiomatic `ExportSchemaResult | undefined`?

Reply:
the motivation was so that existing subclasses returning void don't need to add a `return undefined` to their overrides when consuming this update. I will go test what happens if I make it `| undefined`